src/parser.rs

```diff
@@ -0,0 +1,287 @@
+
+
+pub struct Parser {
+
+}
+
+
+
+impl Parser {
+
+    fn parse_program(&self) {
+        loop {
+            if self.try_peek_eof() { break; }
+
+            self.parse_class_declaration();
+        }
+
+        self.read_eof();
+    }
+
+    fn parse_class_declaration(&self) {
+        self.read_keyword(Keyword.Class);
+        self.read_identifier();
+
+        self.read_op(Operator.LeftBracket);
+        loop {
+            if self.try_peek_op(Operator.RightBracket) { break; }
+
+            self.parse_class_member();
+        }
+        self.read_op(Operator.RightBracket);
+    }
+
+    fn parse_class_member(&self) {
+        self.read_keyword(Keyword.Public);
+
+        self.try_read_keyword(Keyword.Static);
+        self.parse_type();
+        self.read_identifier();
+
+        if self.try_peek(Operator.LeftParen) {
+            self.read(Operator.RightParen);
+
+            // method or main method
```

Comment:
This won't work. We need many more checks for the main method. -> _Is the param `String[] IDENT` the only param_?